.github/workflows/nix-ci.yml

```diff
@@ -0,0 +1,107 @@
+name: Nix CI
+
+on:
+  workflow_dispatch: # allows manual triggering
+  push:
+    branches:
+      - master
+    paths: ['.github/workflows/**', '**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.hpp', '**/*.c', '**/*.cpp', '**/*.cu', '**/*.swift', '**/*.m', '**/*.nix']
```

Comment:
<details><summary>Currently, this excludes the following files.</summary>

```
LICENSE
README.md
SHA256SUMS
awq-py/README.md
awq-py/awq/apply_awq.py
awq-py/requirements.txt
build.zig
ci/README.md
ci/run.sh
cmake/FindSIMD.cmake
codecov.yml
common/build-info.cpp.in
convert-hf-to-gguf.py
convert-llama-ggml-to-gguf.py
convert-lora-to-ggml.py
convert-persimmon-to-gguf.py
convert.py
docs/BLIS.md
docs/llama-star/idea-arch.key
docs/llama-star/idea-arch.pdf
docs/token_generation_performance_tips.md
examples/Miku.sh
examples/alpaca.sh
examples/batched-bench/README.md
examples/batched/README.md
examples/chat-13B.sh
examples/chat-persistent.sh
examples/chat-vicuna.sh
examples/chat.sh
examples/convert-llama2c-to-ggml/README.md
examples/embedding/README.md
examples/export-lora/README.md
examples/finetune/README.md
examples/finetune/convert-finetune-checkpoint-to-gguf.py
examples/finetune/finetune.sh
examples/gpt4all.sh
examples/infill/README.md
examples/jeopardy/README.md
examples/jeopardy/graph.py
examples/jeopardy/jeopardy.sh
examples/jeopardy/qasheet.csv
examples/jeopardy/questions.txt
examples/json-schema-to-grammar.py
examples/llama-bench/README.md
examples/llama.swiftui/README.md
examples/llama.swiftui/llama.swiftui.xcodeproj/project.pbxproj
examples/llama.swiftui/llama.swiftui.xcodeproj/project.xcworkspace/contents.xcworkspacedata
examples/llama.swiftui/llama.swiftui/Assets.xcassets/AccentColor.colorset/Contents.json
examples/llama.swiftui/llama.swiftui/Assets.xcassets/AppIcon.appiconset/Contents.json
examples/llama.swiftui/llama.swiftui/Assets.xcassets/Contents.json
examples/llama.swiftui/llama.swiftui/Preview Content/Preview Assets.xcassets/Contents.json
examples/llama.vim
examples/llama2-13b.sh
examples/llama2.sh
examples/llava/README.md
examples/llava/convert-image-encoder-to-gguf.py
examples/llava/llava-surgery.py
examples/llm.vim
examples/lookahead/README.md
examples/lookup/README.md
examples/main-cmake-pkg/README.md
examples/main/README.md
examples/make-ggml.py
examples/parallel/README.md
examples/perplexity/README.md
examples/quantize/README.md
examples/reason-act.sh
examples/server-llama2-13B.sh
examples/server/README.md
examples/server/api_like_OAI.py
examples/server/chat-llama2.sh
examples/server/chat.mjs
examples/server/chat.sh
examples/server/deps.sh
examples/server/public/completion.js
examples/server/public/index.html
examples/server/public/index.js
examples/server/public/json-schema-to-grammar.mjs
examples/simple/README.md
examples/speculative/README.md
examples/train-text-from-scratch/README.md
examples/train-text-from-scratch/convert-train-checkpoint-to-gguf.py
flake.lock
gguf-py/LICENSE
gguf-py/README.md
gguf-py/examples/writer.py
gguf-py/gguf/__init__.py
gguf-py/gguf/constants.py
gguf-py/gguf/gguf.py
gguf-py/gguf/gguf_reader.py
gguf-py/gguf/gguf_writer.py
gguf-py/gguf/py.typed
gguf-py/gguf/tensor_mapping.py
gguf-py/gguf/vocab.py
gguf-py/pyproject.toml
gguf-py/scripts/__init__.py
gguf-py/scripts/gguf-convert-endian.py
gguf-py/scripts/gguf-dump.py
gguf-py/scripts/gguf-set-metadata.py
gguf-py/tests/test_gguf.py
grammars/README.md
grammars/arithmetic.gbnf
grammars/c.gbnf
grammars/chess.gbnf
grammars/japanese.gbnf
grammars/json.gbnf
grammars/json_arr.gbnf
grammars/list.gbnf
media/llama-leader.jpeg
media/llama0-banner.png
media/llama0-logo.png
media/llama1-banner.png
media/llama1-logo.png
mypy.ini
prompts/LLM-questions.txt
prompts/alpaca.txt
prompts/assistant.txt
prompts/chat-with-baichuan.txt
prompts/chat-with-bob.txt
prompts/chat-with-qwen.txt
prompts/chat-with-vicuna-v0.txt
prompts/chat-with-vicuna-v1.txt
prompts/chat.txt
prompts/dan-modified.txt
prompts/dan.txt
prompts/mnemonics.txt
prompts/parallel-questions.txt
prompts/reason-act.txt
requirements-hf-to-gguf.txt
requirements.txt
run_with_preset.py
scripts/LlamaConfig.cmake.in
scripts/build-info.cmake
scripts/build-info.sh
scripts/convert-gg.sh
scripts/gen-build-info-cpp.cmake
scripts/get-flags.mk
scripts/get-wikitext-2.sh
scripts/qnt-all.sh
scripts/run-all-perf.sh
scripts/run-all-ppl.sh
scripts/server-llm.sh
scripts/sync-ggml-am.sh
scripts/sync-ggml.last
scripts/sync-ggml.sh
scripts/verify-checksum-models.py
tests/test-tokenizer-0-falcon.py
tests/test-tokenizer-0-llama.py
```
</details>

As more Python gets packaged, I think we'll add Python files into this list of paths. Maybe the grammars too.